operator/pkg/ciliumenvoyconfig/watcher.go

```diff
@@ -22,7 +21,7 @@ func StartCECController(ctx context.Context, clientset k8sClient.Clientset, serv
 			log.WithError(err).Fatal("Failed to retrieve service store")
 		}
 
-		m, err := ciliumenvoyconfig.New(ctx, clientset, store.CacheStore(), ports, defaultAlgorithm, idleTimeoutSeconds)
```

Comment:
"as is" migration without optimizing this further. leaving this for a follow up PR. either by optimizing current approach or going with `controller-runtime`